specification/mediaservices/resource-manager/Microsoft.Media/preview/2018-03-30-preview/ContentKeyPolicies.json

```diff
@@ -0,0 +1,1167 @@
+          "description": "The amount of time that the license is valid after the license is first used to play content."
+        },
+        "scmsRestriction": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Configures the Serial Copy Management System (SCMS) in the license. Must be between 0 and 3 inclusive."
+        },
+        "agcAndColorStripeRestriction": {
+          "type": "integer",
+          "format": "int32",
+          "description": "Configures Automatic Gain Control (AGC) and Color Stripe in the license. Must be between 0 and 3 inclusive."
+        },
+        "explicitAnalogTelevisionOutputRestriction": {
+          "$ref": "#/definitions/ContentKeyPolicyPlayReadyExplicitAnalogTelevisionRestriction",
+          "description": "Configures the Explicit Analog Television Output Restriction in the license. Configuration data must be between 0 and 3 inclusive."
+        },
+        "digitalVideoOnlyContentRestriction": {
+          "type": "boolean",
+          "description": "Enables the Image Constraint For Analog Component Video Restriction in the license."
+        },
+        "imageConstraintForAnalogComponentVideoRestriction": {
+          "type": "boolean",
+          "description": "Enables the Image Constraint For Analog Component Video Restriction in the license."
+        },
+        "imageConstraintForAnalogComputerMonitorRestriction": {
+          "type": "boolean",
+          "description": "Enables the Image Constraint For Analog Component Video Restriction in the license."
+        },
+        "allowPassingVideoContentToUnknownOutput": {
+          "type": "string",
+          "enum": [
+            "Unknown",
+            "NotAllowed",
+            "Allowed",
+            "AllowedWithVideoConstriction"
+          ],
+          "x-ms-enum": {
+            "name": "ContentKeyPolicyPlayReadyUnknownOutputPassingOption",
+            "values": [
+              {
+                "value": "Unknown",
+                "description": "Represents a ContentKeyPolicyPlayReadyUnknownOutputPassingOption that is unavailable in current API version."
+              },
+              {
+                "value": "NotAllowed",
+                "description": "Passing the video portion of protected content to an Unknown Output is not allowed."
+              },
+              {
+                "value": "Allowed",
+                "description": "Passing the video portion of protected content to an Unknown Output is allowed."
+              },
+              {
+                "value": "AllowedWithVideoConstriction",
+                "description": "Passing the video portion of protected content to an Unknown Output is allowed but with constrained resolution."
+              }
+            ],
+            "modelAsExtensible": true
+          },
+          "description": "Configures Unknown output handling settings of the license."
+        },
+        "uncompressedDigitalVideoOpl": {
```

Comment:
>uncompressedDigitalVideoOpl [](start = 9, length = 27)

Since `Opl` can be somewhat ambiguous wihtout looking at the description, consider using `x-ms-client-name`

Reply:
OPL stands for output protection level which is a feature of the Microsoft PlayReady DRM system.  This terminology is straight out of the PlayReady compliance guidelines (see section 6.7 of http://download.microsoft.com/download/1/6/D/16D24A3C-8B1F-452A-AACC-C8879D0497DF/Compliance_Rules_For_PlayReady_Products_10_October_2017.docx).  Customers familiar with PlayReady know what an OPL is and will be looking to set it.

Reply:
Okok :)